### ISSUE TITLE
Compiling+installing seems to be problematic on GNU/Linux

### ISSUE THREAD

**Issue by urmum-69** [GRRLIB/GRRLIB#5]
After installing all the needed packages using pacman, installing GRRLIB seems to fail.
```
make[1]: Entering directory '/home/peter/gitrepos/GRRLIB/GRRLIB/GRRLIB'
GRRLIB_3D.c
GRRLIB_3D.c:804:1: fatal error: opening dependency file /GRRLIB_3D.d: Permission denied
  804 | }
      | ^
compilation terminated.
make[1]: *** [/opt/devkitpro/devkitPPC//base_rules:18: GRRLIB_3D.o] Error 1
```
I'm not really sure why this fails, but running it as root seems to work... although that causes the issue of having this later
```
/opt/devkitpro/devkitPPC/bin/../lib/gcc/powerpc-eabi/10.2.0/../../../../powerpc-eabi/bin/ld: cannot open map file 3D_CubedTileDemo.elf.map: Permission denied
```
this does compile fine if I set the needed environment variables and run it as root
I'm probably just being stupid, but I'm not really sure if there is a more appropriate way than just opening an issue on this repo.

**Comment by Crayon2000** [comment]
What exact command did you use? Normally these commands should work without using sudo:
```bash
git clone https://github.com/GRRLIB/GRRLIB.git
cd GRRLIB/GRRLIB/
dkp-pacman --sync --needed --noconfirm libfat-ogc ppc-libpng ppc-freetype ppc-libjpeg-turbo
make clean
make all
make install
```

**Comment by urmum-69** [comment]
here, it fails with 
```
[peter@gamer GRRLIB]$ make all
-----------------------------------------------------------
make[1]: Entering directory '/home/peter/gitrepos/GRRLIB/GRRLIB/GRRLIB'
GRRLIB_3D.c
GRRLIB_3D.c:804:1: fatal error: opening dependency file /GRRLIB_3D.d: Permission denied
  804 | }
      | ^
compilation terminated.
make[1]: *** [/opt/devkitpro/devkitPPC/base_rules:18: GRRLIB_3D.o] Error 1
make[1]: Leaving directory '/home/peter/gitrepos/GRRLIB/GRRLIB/GRRLIB'
make: *** [Makefile:4: all] Error 1
```
on `make all`
I have all needed libs installed using pacman

```
[peter@gamer GRRLIB]$ pacman --sync --needed --noconfirm libfat-ogc ppc-libpng ppc-freetype ppc-libjpeg-turbo
error: you cannot perform this operation unless you are root.
[peter@gamer GRRLIB]$ sudo !!
sudo pacman --sync --needed --noconfirm libfat-ogc ppc-libpng ppc-freetype ppc-libjpeg-turbo
[sudo] password for peter: 
warning: libfat-ogc-1.1.5-2 is up to date -- skipping
warning: ppc-libpng-1.6.37-1 is up to date -- skipping
warning: ppc-freetype-2.9.1-2 is up to date -- skipping
warning: ppc-libjpeg-turbo-1.5.3-4 is up to date -- skipping
 there is nothing to do
```
I'm really not certain what is causing this.

**Comment by Crayon2000** [comment]
Can you try to `git clone` into another folder or remove _/home/peter/gitrepos/GRRLIB_?
Or / and
Make sure you do a `make clean` to remove all _.d_, _.map_ and _.o_ files before the `make all`. These files are generated, so this is weird you get a `Permission denied` on those! If you can write them, you should be able to read them!

**Comment by urmum-69** [comment]
cloning once more results with the same error, running `make all` fails in the same way
I am running `make clean` beforehand.
As a desperate attempt, I tried running chmod 777 on it, that didn't change anything.

**Comment by Crayon2000** [comment]
If nothing is working, well use sudo: `sudo -E make all`

**Comment by urmum-69** [comment]
That all worked.
I couldn't find the true issue, but oh well.
Thanks for your time.

**Comment by ChrisPVille** [comment]
I ran into this as well from a fresh clone and devkitPPC r38-1.  Looks like the *.d dep files are put into $DEPSDIR by the included devkitpro makefile template.  I needed to define ```export DEPSDIR :=      $(CURDIR)/$(BUILD)``` in the GRRLIB Makefile like the devkitpro examples otherwise the dep files were trying to write at the root of the filesystem.

**Comment by Crayon2000** [comment]
@ChrisPVille, thanks for pointing that out. It was fixed in PR #7.
